src/utils/getAreasList/getAreasList.js

```diff
@@ -19,24 +18,31 @@ export type TAreasList = {
   templates: TTemplate[],
 }
 
-export default function getAreasList(props: TProps): TAreasList {
-  const propKeys = Object.keys(props)
+export type TTemplateProps = {
+  [propName: string]: string
+}
 
-  /*
-    if the propValue is not a string it is disqualified from being a template
-    from the very get go, mini-optimization there + it removes the need to
-    type-cast the value passed to sanitizeTemplateString since flow can now
-    infer that the value can only be a string otherwise it would have been
-    filtered out already
-  */
-  const templatePropKeys = propKeys.filter(
-    (propKey) =>
-      typeof props[propKey] === 'string' && /template/i.test(propKey),
-  )
 
-  const areasList = templatePropKeys.reduce(
+/*
+  TODO: rename to genAreasList() ? (for generate)
```

Comment:
We can use `compose`, which I favor for data composition.

Contractions are evil, don't use contractions :)